secrets/api.go

```diff
@@ -0,0 +1,52 @@
+package secrets
+
+import (
+	"github.com/docker/go-plugins-helpers/sdk"
+	"net/http"
+)
+
+const (
+	manifest = `{"Implements": ["secretprovider"]}`
+	getPath  = "/SecretProvider.GetSecret"
+)
+
+// Request is the plugin secret request
+type Request struct {
```

Comment:
In addition to the `Name`, we were considering sending the following information from the service (direct mapping if swarmkit backed):
- `ServiceSpec.TaskSpec.ContainerSpec.Hostname` - for use in a x509 certificate
- `ServiceSpec.EndpointSpec` - has desired DNS domains that might be suitable for an external x509 certificate
- `ServiceSpec.Annotations.Name` - name of the service
- `ServiceSpec.Annotations.Labels` - captures environment names, and other metadata

This can probably be abstracted to:
- Service Hostname
- DNS Domains
- ServiceName
- Metadata (generic map)